compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -489,6 +489,18 @@ object JavaParsers {
             addAnnot(scalaDot(jtpnme.VOLATILEkw))
           case SYNCHRONIZED | STRICTFP =>
             in.nextToken()
+          case SEALED =>
+            flags |= Flags.Sealed
```

Comment:
this is good actually because when we reach TASTy the Child annotations are correctly inferred.